spec/support/tracer_helpers.rb

```diff
@@ -10,43 +10,41 @@ def tracer
   end
 
   def new_tracer(options = {})
-    @tracer ||= begin
```

Comment:
We are overaggressively caching the `new_tracer` in an instance variable, even though the method `tracer` defined just above takes care of that (`def tracer; @tracer ||= new_tracer; end`).

I've made this change to allow multiple calls to `new_tracer` to actually return different instances. This necessary to test that multiple instances of the tracer each have their own thread-local contexts.

No test currently points directly to `new_tracer`, making this change easier.